packages/sync-actions/src/utils/create-build-actions.js

```diff
@@ -1,7 +1,59 @@
+import isEqual from 'lodash.isequal'
+
 function applyOnBeforeDiff(before, now, fn) {
   return fn && typeof fn === 'function' ? fn(before, now) : [before, now]
 }
 
+const createPriceComparator = price => ({
+  value: { currencyCode: price.value.currencyCode },
+  channel: price.channel,
+  country: price.country,
+  customerGroup: price.customerGroup,
+  validFrom: price.validFrom,
+  validUntil: price.validUntil,
+})
+
+function arePricesStructurallyEqual(oldPrice, newPrice) {
+  const oldPriceComparison = createPriceComparator(oldPrice)
+  const newPriceComparison = createPriceComparator(newPrice)
+  return isEqual(newPriceComparison, oldPriceComparison)
+}
+
+function getPriceId(newPrice, previousVariants) {
+  let newPriceId
+
+  previousVariants.forEach(variant => {
+    variant.prices.find(oldPrice => {
+      if (arePricesStructurallyEqual(oldPrice, newPrice)) {
+        newPriceId = oldPrice.id
+        return true
+      }
+      return false
+    })
+  })
+
+  return newPriceId
+}
+
+function updateMissingPriceIds(nextVariants, previousVariants) {
```

Comment:
Maybe rename to `injectMissingPriceIds` or `augmentWithMissingPriceIds`. I know I am strange with naming :)